src/mhd.f90

```diff
@@ -0,0 +1,914 @@
+    !
+    ! FIXME
+
+    real(mytype),dimension(ph1%zst(1):ph1%zen(1),ph1%zst(2):ph1%zen(2),nzmsize) :: div3
+    real(mytype), dimension(xsize(1), xsize(2), xsize(3)) :: ep1
+    real(mytype), dimension(xsize(1), xsize(2), xsize(3), nrhotime) :: rho1
+    real(mytype), dimension(xsize(1), xsize(2), xsize(3), ntime) :: drho1
+    real(mytype), dimension(zsize(1), zsize(2), zsize(3))  :: divu3
+
+    if ((mod(itime,ilist)==0 .or. itime == ifirst .or. itime == ilast)) then
+      call divergence(div3,rho1,Bm(:,:,:,1),Bm(:,:,:,2),Bm(:,:,:,3),ep1,drho1,divu3,2)
+    endif
+
+  end subroutine test_magnetic
+  !
+  subroutine mhd_rhs_eq(dB,B,B0,ux1,uy1,uz1)
+
+    use param
+    use variables
+    use decomp_2d
+    use var, only : ta1,tb1,tc1,td1,te1,tf1,tg1,th1,ti1,di1,mu1,mu2,mu3
+    use var, only : ux2,uy2,uz2,ta2,tb2,tc2,td2,te2,tf2,tg2,th2,ti2,tj2,di2
+    use var, only : ux3,uy3,uz3,ta3,tb3,tc3,td3,te3,tf3,tg3,th3,ti3,di3
+    use var, only : sgsx1,sgsy1,sgsz1
+    use var, only : FTx, FTy, FTz, Fdiscx, Fdiscy, Fdiscz
+    use ibm_param, only : ubcx,ubcy,ubcz
+    use mpi
+
+    implicit none
+
+    !! INPUTS
+    real(mytype),dimension(xsize(1),xsize(2),xsize(3)) :: ux1,uy1,uz1
+
+    !! OUTPUTS
+    real(mytype),dimension(xsize(1),xsize(2),xsize(3),1:3) :: B,B0,dB
+    
+    integer :: i,j,k,is
+    !
+
+    ! local 
+    real(mytype) :: rrem
+    real(mytype), save, allocatable, dimension(:,:,:) :: tx1,ty1,tz1,tx2,ty2,tz2, &
+                                                         tx3,ty3,tz3,bx2,by2,bz2, &
+                                                         bx3,by3,bz3,cx2,cy2,cz2, &
+                                                         cx3,cy3,cz3
+    real(mytype),dimension(xsize(1),xsize(2),xsize(3),1:3) :: Bsum
+    logical,save :: firstcal=.true.
+
+    if(firstcal) then
+      !
+      call alloc_x(tx1)
+      tx1 = zero
+      call alloc_x(ty1)
+      ty1 = zero
+      call alloc_x(tz1)
+      tz1 = zero
+  
+      call alloc_y(tx2)
+      tx2=zero
+      call alloc_y(ty2)
+      ty2=zero
+      call alloc_y(tz2)
+      tz2=zero
+  
+      call alloc_y(bx2)
+      bx2=zero
+      call alloc_y(by2)
+      by2=zero
+      call alloc_y(bz2)
+      bz2=zero
+
+      call alloc_z(tx3)
+      tx3=zero
+      call alloc_z(ty3)
+      ty3=zero
+      call alloc_z(tz3)
+      tz3=zero
+      !
+      call alloc_z(bx3)
+      bx3=zero
+      call alloc_z(by3)
+      by3=zero
+      call alloc_z(bz3)
+      bz3=zero
+      !
+      call alloc_y(cx2)
+      bx2=zero
+      call alloc_y(cy2)
+      by2=zero
+      call alloc_y(cz2)
+      bz2=zero
+      !
+      call alloc_z(cx3)
+      bx3=zero
+      call alloc_z(cy3)
+      by3=zero
+      call alloc_z(cz3)
+      bz3=zero
+      !
+      firstcal=.false.
+    endif
+
+    rrem=1._mytype/Rem
+
+    Bsum=B+B0
+
+    !SKEW SYMMETRIC FORM
+
+    !WORK X-PENCILS
+    ta1(:,:,:) = ux1(:,:,:) * Bsum(:,:,:,1) - Bsum(:,:,:,1) * ux1(:,:,:) ! FIXME this is always zero
```

Comment:
@rfj82982 ?

Reply:
See the math attached and yes this is zero. We could expand all the terms and remove all the bits that are zero. cid:3292591482*5c6baf65-e13e-45ff-a40a-eb88b6d13c15

Reply:
If it is always zero, can we write the following ?
```
ta1(:,:,:) = 0._mytype ! ux1(:,:,:) * Bsum(:,:,:,1) - Bsum(:,:,:,1) * ux1(:,:,:) is always zero
```

Reply:
![thumbnail_image](https://github.com/xcompact3d/Incompact3d/assets/35581400/7c60b466-54f1-4db7-9ddf-749e632eb4ae)

Reply:
As you can see the terms on the diagonal are 0

Reply:
basically ta1, te2 and tf3 should be zero

Reply:
Can we skip their estimation and their derivation then ?

Reply:
I think so, just need to be carefully. I'll try to have a go and hopefully i won't remove a non zero term

Reply:
Ok I have set to zero the diagonal terms as well as the derivative. I kept the terms in the sum even if zero for consistency.

Reply:
Results looks ok